python/dolfinx/io/ngsio.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (C) 2022 Jørgen S. Dokken
+#
+# This file is part of DOLFINx (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+"""Tools to extract data from Gmsh models"""
+import typing
+
+import numpy as np
+import numpy.typing as npt
+
+import basix
+import basix.ufl
+import ufl
+from dolfinx import cpp as _cpp
+from dolfinx.cpp.graph import AdjacencyList_int32
+from dolfinx.mesh import (CellType, GhostMode, Mesh, create_cell_partitioner,
+                          create_mesh, meshtags, meshtags_from_entities)
+
+from mpi4py import MPI as _MPI
+
+__all__ = ["cell_perm_array", "ufl_mesh"]
+
+try:
+    import ngsolve as ngs
+    from ngsPETSc.plex import DMPlexMapping
+    import netgen as ng
+    _has_ngs = True
+except ModuleNotFoundError:
+    _has_ngs = False
+
+
+# Map from Gmsh cell type identifier (integer) to DOLFINx cell type
+# and degree http://gmsh.info//doc/texinfo/gmsh.html#MSH-file-format
+_ngs_to_cells = {(2,3): ("triangle", 1),
+                 (2,4): ("quadrilateral", 1),
+                 (3,4): ("tetrahedron", 1)}
+
+def ufl_mesh(ngs_cell: int, gdim: int) -> ufl.Mesh:
+    """Create a UFL mesh from a Gmsh cell identifier and the geometric dimension.
+
+    See: http://gmsh.info//doc/texinfo/gmsh.html#MSH-file-format.
+
+    Args:
+        gmsh_cell: The Gmsh cell identifier
+        gdim: The geometric dimension of the mesh
+
+    Returns:
+        A ufl Mesh using Lagrange elements (equispaced) of the
+        corresponding DOLFINx cell
+    """
+    shape, degree = _ngs_to_cells[ngs_cell]
+    print("Shape: {}, Degree: {}".format(shape,degree))
+    cell = ufl.Cell(shape, geometric_dimension=gdim)
+
+    element = basix.ufl.element(
+        basix.ElementFamily.P, cell.cellname(), degree, basix.LagrangeVariant.equispaced, shape=(gdim, ),
+        gdim=gdim)
+    return ufl.Mesh(element)
+
+
+def cell_perm_array(cell_type: CellType, num_nodes: int) -> typing.List[int]:
```

Comment:
Do we need to expose this function,or should it just be used with `_cpp.perm_gmsh`?

Reply:
No need to expose this function you are right. In particular, considering that it's already exposed in the io.gmsh